opencti-platform/opencti-graphql/src/manager/ingestionManager.ts

```diff
@@ -204,59 +204,106 @@ const rssExecutor = async (context: AuthContext, turndownService: TurndownServic
 // endregion
 
 // region Taxii ingestion
-interface TaxiiResponseData {
-  data: { more: boolean, next: string, objects: object[] },
-  addedLast: string | undefined | null
+export interface TaxiiResponseData {
+  data: { more: boolean | undefined, next: string | undefined, objects: object[] },
+  addedLastHeader: string | undefined | null
 }
 
+/**
+ *  Compute HTTP GET parameters to send to taxii server.
+ *
+ *  @see https://docs.oasis-open.org/cti/taxii/v2.1/os/taxii-v2.1-os.html#_Toc31107519
+ *   // If the more property is set to true and the next property is populated
+ *   // then the client can paginate through the remaining records
+ *   // using the next URL parameter along with the same original query options.
+ *   // If the more property is set to true and the next property is empty
+ *   // then the client may paginate through the remaining records by using the added_after URL parameter with the
+ *   // date/time value from the X-TAXII-Date-Added-Last header along with the same original query options.
+ * @param ingestion
+ */
+export const prepareTaxiiGetParam = (ingestion: BasicStoreEntityIngestionTaxii) => {
+  const next = ingestion.current_state_cursor;
+  const added_after = ingestion.added_after_start;
+  const more = ingestion.taxii_more || false;
+  if (more) {
```

Comment:
why do we need this condition ? since we reset next to undefined when more is false, we could just send the two params, right ?